diam/reflect.go

```diff
@@ -368,6 +378,14 @@ func scanStruct(m *Message, field reflect.Value, avps []*AVP) error {
 	for n := 0; n < base.NumField(); n++ {
 		f := base.Field(n)
 		bt := base.Type().Field(n)
+
+		if bt.Anonymous {
+			if err := scanStruct(m, f, avps); err != nil {
+				return err
+			}
+			continue
+		}
+
```

Comment:
It will fail if there is non struct embedded field. Something like this could work:

`
		if bt.Anonymous  && bt.Type.Kind() == reflect.Struct && len(bt.Tag) == 0 {
			if err := scanStruct(m, f, avps); err != nil {
				return err
			}
			continue
		}

`

Reply:
Fixed in 061a6ec780e303403aa849cf71e3dc3762258b5d